spec/migrations/20171026103833_add_cores_allocated_rate_detail_spec.rb

```diff
@@ -0,0 +1,46 @@
+require_migration
+
+describe AddCoresAllocatedRateDetail do
+  let(:rate_stub) { migration_stub(:ChargebackRate) }
+  let(:detail_stub) { migration_stub(:ChargebackRateDetail) }
+  let(:field_stub) { migration_stub(:ChargeableField) }
+  let(:measure_stub) { migration_stub(:ChargebackRateDetailMeasure) }
+
+  let(:default_rate) { rate_stub.create!(:rate_type => 'Compute', :default => :true) }
+  let(:custom_rate) { rate_stub.create!(:rate_type => 'Compute') }
+
+  let(:rate_details) do
+    [
+      {:metric => 'cpu_usagemhz_rate_average', :group => 'cpu', :source => 'used', :description => 'Used CPU'},
+      {:metric => 'derived_vm_numvcpus', :group => 'cpu', :source => 'allocated', :description => 'Allocated CPU Count'},
+      {:metric => 'disk_usage_rate_average', :group => 'disk_io', :source => 'used', :description => 'Used Disk I/O'}
+    ]
+  end
+
+  let(:allocated_cores) { {:metric => 'derived_vm_numvcpus_cores', :group => 'cpu cores', :source => 'allocated', :description => 'Allocated CPU Cores'} }
+
+  migration_context :up do
+    it 'Adds a "Allocated CPU Cores" rate detail to existing details' do
+      rate_details.each do |field|
+        detail_stub.create!(field.merge(:chargeback_rate_id => default_rate.id))
+        detail_stub.create!(field.merge(:chargeback_rate_id => custom_rate.id))
+      end
+
+      migrate
+
+      expect(default_rate.chargeback_rate_details.where(:description => "Allocated CPU Cores").count).to eq(1)
```

Comment:
Since the default_rate object is memoized from the call on line 25, can you change this to a `default_rate.reload.chargeback_rate_details...`?   This way we can ensure it's getting fresh data.